src/activities/assignments/AssignmentEntity.js

```diff
@@ -554,6 +563,30 @@ export class AssignmentEntity extends Entity {
 		await performSirenAction(this._token, action, fields);
 	}
 
+	/**
+	 * Sets the custom allowable filetypes of the assignment
+	 * @param {string} customAllowableFileTypes Allowable filetype option
+	 */
+	async setCustomAllowableFileTypes(customAllowableFileTypes) {
+		customAllowableFileTypes = String(customAllowableFileTypes);
+		const action = this.canEditAllowableFileType() && this._entity.getActionByName(Actions.assignments.updateCustomAllowableFileType);
```

Comment:
should we instead be checking if it has the `update-custom-allowable-file-type` action?

Reply:
Changed!